### PR TITLE
[STONEBLD-818] Do not set default component image repository

### PR DESCRIPTION
**Pull request by mmorhun**
### What does this PR do?:
Removes code that sets default value into `ContainerImage` field.
It should be done by user or image controller should provide annotation with the created image repository

### Which issue(s)/story(ies) does this PR fixes:
[STONEBLD-818](https://issues.redhat.com/browse/STONEBLD-818)

### Blocked by:
[HAC-3275](https://issues.redhat.com/browse/HAC-3275)

### CODE REVIEW

**Comment by mmorhun** [comment]
I see the following error in the logs right after component creation:
```
2023-03-20T11:39:15.766Z	ERROR	controllers.Component	Unable to update Component	{"appstudio-component": "HAS", "Component": "go-component-sample", "error": "Operation cannot be fulfilled on components.appstudio.redhat.com \"go-component-sample\": the object has been modified; please apply your changes to the latest version and try again"}
github.com/redhat-appstudio/application-service/controllers.(*ComponentReconciler).Reconcile
	/workspace/controllers/component_controller.go:416
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
	/go/pkg/mod/github.com/kcp-dev/controller-runtime@v0.12.2-0.20220808200255-4b60fd66e5de/pkg/internal/controller/controller.go:121
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/go/pkg/mod/github.com/kcp-dev/controller-runtime@v0.12.2-0.20220808200255-4b60fd66e5de/pkg/internal/controller/controller.go:320
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/github.com/kcp-dev/controller-runtime@v0.12.2-0.20220808200255-4b60fd66e5de/pkg/internal/controller/controller.go:273
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/go/pkg/mod/github.com/kcp-dev/controller-runtime@v0.12.2-0.20220808200255-4b60fd66e5de/pkg/internal/controller/controller.go:234
2023-03-20T11:39:15.771Z	ERROR	controllers.Component	Unable to update Component	{"appstudio-component": "HAS", "Component": "test/go-component-sample", "clusterName": "", "error": "Operation cannot be fulfilled on components.appstudio.redhat.com \"go-component-sample\": the object has been modified; please apply your changes to the latest version and try again"}
github.com/redhat-appstudio/application-service/controllers.(*ComponentReconciler).Reconcile
	/workspace/controllers/component_controller.go:420
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
	/go/pkg/mod/github.com/kcp-dev/controller-runtime@v0.12.2-0.20220808200255-4b60fd66e5de/pkg/internal/controller/controller.go:121
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/go/pkg/mod/github.com/kcp-dev/controller-runtime@v0.12.2-0.20220808200255-4b60fd66e5de/pkg/internal/controller/controller.go:320
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/github.com/kcp-dev/controller-runtime@v0.12.2-0.20220808200255-4b60fd66e5de/pkg/internal/controller/controller.go:273
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/go/pkg/mod/github.com/kcp-dev/controller-runtime@v0.12.2-0.20220808200255-4b60fd66e5de/pkg/internal/controller/controller.go:234
2023-03-20T11:39:15.771Z	INFO	controllers.Component	Finished reconcile loop for test/go-component-sample	{"appstudio-component": "HAS", "Component": "test/go-component-sample", "clusterName": ""}
```
Nothing is set in the status.

File: main.go [hunk 1]
```diff
@@ -168,13 +167,6 @@ func main() {
 		ghOrg = "redhat-appstudio-appdata"
 	}
 
-	// Retrieve the name of the default repository to use
```
Comment (maysunfaisal) [inline main.go:171]: Pls update the README to reflect this ie; it needs to be removed

Comment (mmorhun) [reply main.go:171]: done, please check

File: README.md [hunk 2]
```diff
@@ -55,11 +55,10 @@ The following section outlines the steps to deploy HAS on a physical Kubernetes
...
-`Secrets` containing registry credentials.
-* Configure the default image repository to which Pipelines would push images to by defining the environment variable `IMAGE_REPOSITORY` for the operator deployment.
-Defaults to `quay.io/redhat-appstudio/user-workload`.
+To use auto generated image repository for the Component's image add `image.redhat.com/generate: "true"` annotation to the Component.
+
+If you wish to get 'working' PipelineRuns with user provided image repository, create an image pull secret named `redhat-appstudio-registry-pull-secret` and link it to the `pipeline` Service Account in the Component's namespace (in both `secrets` and `imagePullSecrets` sections).
+See [Kubernetes docs](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#registry-secret-existing-credentials) for more information on how to create `Secrets` containing registry credentials.
 
 Pipelines would use the credentials in the image pull secret `redhat-appstudio-registry-pull-secret` to push to $IMAGE_REPOSITORY.
```
[the remark this answers is not in the document]
Comment (mmorhun) [reply README.md:63]: agree, thanks

File: README.md [hunk 3]
```diff
@@ -55,11 +55,10 @@ The following section outlines the steps to deploy HAS on a physical Kubernetes
...
 As a user, upon creation of Component, Tekton resources would be created by the controller.
 
-If you wish to get 'working' PipelineRuns,
-* Create an image pull secret named `redhat-appstudio-registry-pull-secret`. See [Kubernetes docs](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#registry-secret-existing-credentials) for more information on how to create
-`Secrets` containing registry credentials.
-* Configure the default image repository to which Pipelines would push images to by defining the environment variable `IMAGE_REPOSITORY` for the operator deployment.
-Defaults to `quay.io/redhat-appstudio/user-workload`.
+To use auto generated image repository for the Component's image add `image.redhat.com/generate: "true"` annotation to the Component.
+
+If you wish to get 'working' PipelineRuns with user provided image repository, create an image pull secret named `redhat-appstudio-registry-pull-secret` and link it to the `pipeline` Service Account in the Component's namespace (in both `secrets` and `imagePullSecrets` sections).
```
[the remark this answers is not in the document]
Comment (mmorhun) [reply README.md:60]: ok

**Review by maysunfaisal — APPROVED** [review]

**Comment by maysunfaisal** [comment]
@mmorhun Regd https://github.com/redhat-appstudio/image-controller/pull/13, in application-service, atleast in Application and Component controllers we need to patch the status when updating it rather than doing a direct update. CDQ controller does this to resolve this issue.

We should keep an eye on it since any other controllers can update the status. Is the image-controller status updates in e2e test suite? How is it breaking here?

**Comment by mmorhun** [comment]
@maysunfaisal Image Controller nor Build Service doesn't update the status of the Component. Both of them use annotations. Status was not involved at all. The error described above happens all the time if image controller (but could be any other service) updates the component CR (adds / changes an annotation in the specific case, but I think any update will have such effect) while Application Service is working on setting devfile.
The problem, I think, that Application Service doesn't retry the operation if update failed due to out of date Component config in memory. That's the first way. Another way is to just reread the component before update, e.g.:
```
component := r.Client.Get()
// ...
// complicated logic here
// ...
component := r.Client.Get()
// apply updates calculated before
r.Client.Update(component)
```

**Comment by mmorhun** [comment]
Rebased

**Review by Michkov — APPROVED** [review]

**Pull request merged**